src/gp_engine.cpp

```diff
@@ -100,11 +131,13 @@ void GPEngine::SetTransitionAndDerivativeMatricesToHaveBranchLength(
   Eigen::Vector4d diagonal_vector = (branch_length * eigenvalues_).array().exp();
   diagonal_matrix_.diagonal() = diagonal_vector;
   transition_matrix_ = eigenmatrix_ * diagonal_matrix_ * inverse_eigenmatrix_;
-  diagonal_matrix_.diagonal() = diagonal_vector.array() * eigenvalues_.array();
+  diagonal_matrix_.diagonal() = eigenvalues_.array() * diagonal_vector.array();
```

Comment:
Interesting. I'm not sure if I understand the difference here.

Reply:
No difference! It's just my obsession... in my derivation, I put the eigenvalues vector first. Happy to change it back.